src/settings.rs

```diff
@@ -94,6 +94,55 @@ pub struct CommonSettings {
     )]
     pub nix_build_group_id: u32,
 
+    /// The Nix build user prefix (user numbers will be postfixed)
+    #[cfg_attr(
+        feature = "cli",
+        clap(long, env = "NIX_INSTALLER_NIX_BUILD_USER_PREFIX", global = true)
+    )]
+    #[cfg_attr(
+        all(target_os = "macos", feature = "cli"),
+        clap(default_value = "_nixbld")
+    )]
+    #[cfg_attr(
+        all(target_os = "linux", feature = "cli"),
+        clap(default_value = "nixbld")
+    )]
+    pub nix_build_user_prefix: String,
```

Comment:
I would expect this to not be necessary for Linux with auto uids enabled... Am I wrong?

By that I mean I would expect this to be `None` on Linux and `Some(...)` on Darwin (maybe I misunderstand how we do auto uids though).

Same for all the other settings below.

Reply:
We do set the users to 0, I could move these settings to Mac only settings, and have these actions take that... But I feel like it's entirely possible we'll remove the feature on Linux still so I don't want to possibly make a bunch of work for ourselves later.